webapi/src/main/scala/org/knora/webapi/messages/admin/responder/valueObjects/ProjectsValueObjectsADM.scala

```diff
@@ -0,0 +1,189 @@
+package org.knora.webapi.messages.admin.responder.valueObjects
+
+import org.knora.webapi.exceptions.BadRequestException
+import org.knora.webapi.messages.StringFormatter
+import org.knora.webapi.messages.admin.responder.projectsmessages.ProjectsErrorMessagesADM._
+import org.knora.webapi.messages.store.triplestoremessages.StringLiteralV2
+import zio.prelude.Validation
+
+/**
+ * ProjectIRI value object.
+ */
+sealed abstract case class ProjectIRI private (value: String)
+object ProjectIRI { self =>
+  val sf = StringFormatter.getGeneralInstance
+
+  def make(value: String): Validation[Throwable, ProjectIRI] =
+    if (value.isEmpty) {
+      Validation.fail(BadRequestException(PROJECT_IRI_MISSING_ERROR))
+    } else {
+      if (value.nonEmpty && !sf.isKnoraProjectIriStr(value)) {
+        Validation.fail(BadRequestException(PROJECT_IRI_INVALID_ERROR))
+      } else {
+        val validatedValue = Validation(
+          sf.validateAndEscapeProjectIri(value, throw BadRequestException(PROJECT_IRI_INVALID_ERROR))
+        )
+
+        validatedValue.map(new ProjectIRI(_) {})
+      }
+    }
+
+  def make(value: Option[String]): Validation[Throwable, Option[ProjectIRI]] =
+    value match {
+      case Some(v) => self.make(v).map(Some(_))
+      case None    => Validation.succeed(None)
+    }
+}
+
```

Comment:
To me, this doesn't fully make sense (as I messaged you separately).

I think `make(None)` should only be a success, if the property is optional.  
Also, the reason for the second `make()` method is to be able to both handle `String` and `Option[String]`elegantely, which makes sense to me. But wouldn't this be a good place to get rid of the option? If it's `Some(v)`, you just create it for `v` (and drop the Option); and if it's None, it should imo fail anyways.

But maybe I just don't understand the reasoning behind this.

Reply:
I see your point regarding the second `make()`, but I thought the creation of value objects is at the outermost edge of the API where the actual interaction with the client takes place. And there I thought it's always Strings and Booleans, never Options. But I could be wrong of course.

Reply:
Explained [here](https://github.com/dasch-swiss/dsp-api/pull/1976#discussion_r784655179) and [here](https://github.com/dasch-swiss/dsp-api/pull/1976#discussion_r784657235).